pkg/healthcheck/checkers.go

```diff
@@ -118,6 +118,37 @@ func CheckK8sPods(ctx context.Context, client *kubernetes.Clientset, label strin
 	}
 }
 
+// CheckRedisPort returns a checker which verifies if the default port of redis (6379) is already binded
+// on localhost. If it is, it fails. If not, it succeeds.
+func CheckRedisPort(ctx context.Context, ow *rpc.OutputWriter, cli *client.Client) Checker {
+	return func() (bool, string, error) {
+		// First, we check if we can bind locally to 127.0.0.1:6379
+		ln, err := net.Listen("tcp", "127.0.0.1:6379")
+		if err != nil {
+			return false, "local port 6379 is already occupied; please stop any local Redis instances first.", nil
+		}
+		_ = ln.Close()
```

Comment:
I think this is redundant. We are checking all interfaces below, so why do we need to check 127.0.0.1 specifically?

Also if `testground-redis` is running we really don't need to check anything, no?

Reply:
I'm not sure of the behaviour. Imagine this situation:

1. `testground-redis` is running on all interfaces.
2. A local server is running on `127.0.0.1`.

When we run with `local:exec` we try to connect to `localhost:6379`. Which of those two options does it pick?

Checking if `testground-redis` is running is not enough because it still might connect to some other redis instance as described on #959. If there's evidence it's not like that, I can remove that bit.

Reply:
Right, good catch, I see your point. I honestly don't know if we are not missing a case, but this is better than the current state, so feel free to merge :)

Reply:
This is just another example that supporting so many runners is somewhat difficult, considering that we want to replicate the experience and behaviour across runners (influx, grafana, jupyter, (previously there were some attempts to also include prometheus) and what-not), which is especially difficult when you have no isolation on the machine... at least with Docker we get that isolation.

---

We might be more successful if we just allow Docker to bind all auxiliary services to random ports, and not lock static ports - then I think Docker makes sure there are no collisions. We would just need to introduce code for service discovery though.